### PR TITLE
Fixed interceptor, remove business logic from controller, create enum…

### PR DESCRIPTION
**Pull request by eugent482**
… for statuses and move it to model,reduce complexity

### CODE REVIEW

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs [hunk 1]
```diff
@@ -44,92 +42,56 @@ public IActionResult GetAvailableAppointments([FromQuery] int doctorId)
...
-            }
-            else
+            if (!int.TryParse(User.Identity.Name, out var doctorId))
             {
                 return BadRequest(new { message = "Wrong claims" });
             }
+
+            var (status, message) = _appointmentService.AddAppointment(creationModel, doctorId);
+
+            return status ? (IActionResult)Ok(new { message }) : BadRequest(new { message });
```
Comment (TREN-D) [inline InternetHospital.WebApi/Controllers/AppointmentsController.cs:52]: Why you used explicit conversion to IActionResult and only for Ok()?

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs [hunk 2]
```diff
@@ -44,92 +42,56 @@ public IActionResult GetAvailableAppointments([FromQuery] int doctorId)
...
-            }
-            else
+            if (!int.TryParse(User.Identity.Name, out var doctorId))
             {
                 return BadRequest(new { message = "Wrong claims" });
             }
+
+            var (status, message) = _appointmentService.CancelAppointment(model.Id, doctorId);
+
+            return status ? (IActionResult)Ok(new { message }) : BadRequest(new { message });
```
Comment (TREN-D) [inline InternetHospital.WebApi/Controllers/AppointmentsController.cs:66]: Here the same question.

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs [hunk 3]
```diff
@@ -44,92 +42,56 @@ public IActionResult GetAvailableAppointments([FromQuery] int doctorId)
...
-            }
-            else
+            if (!int.TryParse(User.Identity.Name, out var doctorId))
             {
                 return BadRequest(new { message = "Wrong claims" });
             }
+
+            var (status, message) = _appointmentService.DeleteAppointment(model.Id, doctorId);
+
+            return status ? (IActionResult)Ok(new { message }) : BadRequest(new { message });
```
Comment (TREN-D) [inline InternetHospital.WebApi/Controllers/AppointmentsController.cs:80]: Here the same question.

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs [hunk 4]
```diff
@@ -44,92 +42,56 @@ public IActionResult GetAvailableAppointments([FromQuery] int doctorId)
...
-            }
-            else
+            if (!int.TryParse(User.Identity.Name, out var doctorId))
             {
                 return BadRequest(new { message = "Wrong claims" });
             }
+
+            var (status, message) = _appointmentService.FinishAppointment(model.Id, doctorId);
+
+            return status ? (IActionResult)Ok(new { message }) : BadRequest(new { message });
```
Comment (TREN-D) [inline InternetHospital.WebApi/Controllers/AppointmentsController.cs:94]: Here the same question.

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs (hunk 1 above)
Comment (eugent482) [reply InternetHospital.WebApi/Controllers/AppointmentsController.cs:52]: There is no implicit conversion between OkObjectResult and BadRequestObjectResult

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs (hunk 2 above)
Comment (eugent482) [reply InternetHospital.WebApi/Controllers/AppointmentsController.cs:66]: Here the same answer: There is no implicit conversion between OkObjectResult and BadRequestObjectResult

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs (hunk 3 above)
Comment (eugent482) [reply InternetHospital.WebApi/Controllers/AppointmentsController.cs:80]: Here the same answer: There is no implicit conversion between OkObjectResult and BadRequestObjectResult

File: InternetHospital.WebApi/Controllers/AppointmentsController.cs (hunk 4 above)
Comment (eugent482) [reply InternetHospital.WebApi/Controllers/AppointmentsController.cs:94]: Here the same answer: There is no implicit conversion between OkObjectResult and BadRequestObjectResult

**Comment by TREN-D** [comment]
Is it okay to return Tuples instead of ViewModel?

**Review by SergiyKozachok — APPROVED** [review]

**Review by serhiidotsyk — APPROVED** [review]

**Review by TREN-D — APPROVED** [review]

**Pull request merged**